test/gurka/test_ramps_tc.cc

```diff
@@ -290,8 +290,115 @@ void check_edge_classification(baldr::GraphReader& graph_reader,
   const auto edge = std::get<1>(gurka::findEdgeByNodes(graph_reader, nodes, b, e));
   EXPECT_EQ(edge->classification(), rc);
 }
+
+void check_edge_use(baldr::GraphReader& graph_reader,
+                    const gurka::nodelayout& nodes,
+                    const std::string& b,
+                    const std::string& e,
+                    baldr::Use tc) {
+  const auto edge = std::get<1>(gurka::findEdgeByNodes(graph_reader, nodes, b, e));
+  EXPECT_EQ(edge->use(), tc);
+}
+
 } // namespace
 
+TEST(RampsNoReclass, test_tc_infer) {
+
+  constexpr double gridsize_metres = 10;
+
+  const std::string ascii_map = R"(
+
+      A---B-------------------------------C------D
+           \                             /
+            \            F J            /
+             \           | |           /
+              \          | |          /
+               \         | |         /
+                \        | |        /
+                 --E-----G-K-----N--
+                    \    | |    /
+                     \   | |   /
+                      \  | |  /
+                       \ | | /
+                         H L
+                         | |
+                         | |
+                         | |
+                         I M
+
+
+)";
+
+  const gurka::ways ways = {
+      {"AB", {{"highway", "motorway"}, {"oneway", "yes"}}},
+      {"BC", {{"highway", "motorway"}, {"oneway", "yes"}}},
+      {"CD", {{"highway", "motorway"}, {"oneway", "yes"}}},
+
+      {"BE", {{"highway", "motorway_link"}, {"oneway", "yes"}}},
+      {"EG", {{"highway", "motorway_link"}, {"oneway", "yes"}}},
+      {"EH", {{"highway", "motorway_link"}, {"oneway", "yes"}}},
+      {"GK", {{"highway", "motorway_link"}, {"oneway", "yes"}}},
+      {"FG", {{"highway", "primary"}, {"oneway", "yes"}}},
+      {"GH", {{"highway", "primary"}, {"oneway", "yes"}}},
+      {"HI", {{"highway", "primary"}, {"oneway", "yes"}}},
+      {"ML", {{"highway", "primary"}, {"oneway", "yes"}}},
+      {"LK", {{"highway", "primary"}, {"oneway", "yes"}}},
+      {"KJ", {{"highway", "primary"}, {"oneway", "yes"}}},
+      {"KN", {{"highway", "motorway_link"}, {"oneway", "yes"}}},
+      {"LN", {{"highway", "motorway_link"}, {"oneway", "yes"}}},
+      {"NC", {{"highway", "motorway_link"}, {"oneway", "yes"}}},
+  };
+
+  const gurka::nodes nodes = {{"B", {{"highway", "motorway_junction"}, {"ref", "4"}}}};
+
+  const auto layout = gurka::detail::map_to_coordinates(ascii_map, gridsize_metres);
+  auto map = gurka::buildtiles(layout, ways, {}, {}, "test/data/gurka_ramps_tc_infer",
```

Comment:
when copying from other tests it might be a good idea to use a different dir. in this case its not a problem because the tests are run synchronously but if you had copied from another test in another file, then we can get random failures when the tests are run in parallel (since they race over messing with the same directory)

Reply:
fixed